plutus-core/cost-model/create-cost-model/CreateBuiltinCostModel.hs

```diff
@@ -235,241 +228,217 @@ createBuiltinCostModel bmfile rfile = do
-  -- (max x y) + 1
-  let memModel = ModelTwoArgumentsMaxSize $ ModelMaxSize 1 1
-  pure $ CostingFun cpuModel memModel
+addInteger :: MakeCostingFun2
+addInteger =
+  makeTwoVariableCostingFunction $ ModelTwoArgumentsMaxSize $ OneVariableLinearFunction 1 1
 
-subtractInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-subtractInteger cpuModelR = do
-  cpuModel <- ModelTwoArgumentsMaxSize <$> readModelMaxSize cpuModelR
+subtractInteger :: MakeCostingFun2
+subtractInteger =
+  makeTwoVariableCostingFunction $ ModelTwoArgumentsMaxSize $ OneVariableLinearFunction 1 1
   -- The worst case is subtracting e.g. `minBound :: Int` - `maxBound :: Int`, which increases the memory usage by one.
   -- (max x y) + 1
-  let memModel = ModelTwoArgumentsMaxSize $ ModelMaxSize 1 1
-  pure $ CostingFun (cpuModel) memModel
 
-multiplyInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-multiplyInteger cpuModelR = do
-  cpuModel <- ModelTwoArgumentsAddedSizes <$> readModelAddedSizes cpuModelR
+multiplyInteger :: MakeCostingFun2
+multiplyInteger =
   -- GMP requires multiplication (mpn_mul) to have x + y space.
   -- x + y
-  let memModel = ModelTwoArgumentsAddedSizes $ ModelAddedSizes 0 1
-  pure $ CostingFun (cpuModel) memModel
-
-divideInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-divideInteger cpuModelR = do
-  cpuModelBelowDiag <- readModelMultipliedSizes cpuModelR
-  let cpuModel = ModelTwoArgumentsConstAboveDiagonal
-                 (ModelConstantOrTwoArguments 196500 $
-                  ModelTwoArgumentsMultipliedSizes cpuModelBelowDiag
-                  -- ### FIXME: the constant above is currently obtained manually from R; automate this
-                 )
-  -- GMP requires division (mpn_divrem) to have x - y space.
-  -- x - y
-  let memModel = ModelTwoArgumentsSubtractedSizes $ ModelSubtractedSizes 0 1 1
-  pure $ CostingFun cpuModel memModel
-
-quotientInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
-quotientInteger cpuModelR = do
-  cpuModelBelowDiag <- readModelMultipliedSizes cpuModelR
-  let cpuModel = ModelTwoArgumentsConstAboveDiagonal
-                 (ModelConstantOrTwoArguments 196500 $
-                  ModelTwoArgumentsMultipliedSizes cpuModelBelowDiag
-                  -- ### FIXME: the constant above is currently obtained manually from R; automate this
-                 )
-  -- GMP requires division (mpn_divrem) to have x - y space.
-  -- x - y
-  let memModel = ModelTwoArgumentsSubtractedSizes $ ModelSubtractedSizes 0 1 1
-  pure $ CostingFun cpuModel memModel
-
-remainderInteger :: MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelTwoArguments)
+  makeTwoVariableCostingFunction $ ModelTwoArgumentsAddedSizes $ OneVariableLinearFunction 0 1
+
+divideInteger :: MakeCostingFun2
+divideInteger =
+  makeTwoVariableCostingFunction $ ModelTwoArgumentsSubtractedSizes $ ModelSubtractedSizes 0 1 1
```

Comment:
Wait, but we have 

```haskell
    "subtracted_sizes"     -> ModelTwoArgumentsSubtractedSizes    <$> error "subtracted sizes"
```

?

Reply:
That's the memory usage: we don't need to parse that.

Reply:
I'm not sure I understand, but I trust you!